### DISCUSSION TITLE
How can I build code navigation database for the whole project?

### DISCUSSION THREAD

**Discussion by ltang666** [microsoft/vscode-cpptools#6998] [General]
If a file in the project didn't open, symbols in the file cannot be found when finding some c++ symbols' definitions.
I want to build the whole project navigation database, then all definitions can be found.
How can I do that?

installed c/c++ extension.

Version: 1.53.2
Commit: 622cb03f7e070a9670c94bae1a45d78d7181fbd4
Date: 2021-02-11T11:45:54.515Z
Electron: 11.2.1
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin x64 18.7.0

**Comment by sean-mcmanus** [comment]
What does "a file in the project didn't open mean"? By default, all the files under a workspace folder should be parsed for global symbols. When you open the file with the symbol definition, does the symbol appear in the Outline view? Is the symbol global or local?

What version of the C/C++ extension do you have?

**Reply by ltang666** [reply]
"a file in the project didn't open" means a file never be opened.
c/c++ extension version:  v1.2.1
Came across this situation many times. sometimes when I open a new file, the file function symbols cannot be built. But after I close vscode, then open vscode again, symbols can be built.

**Reply by sean-mcmanus** [reply]
It sounds like you may be hitting a bug, but we'd probably need more repro info in order to fix it.

**Reply by ltang666** [reply]
when I find a definition that shows ’Loading‘ for minutes as follows.  after I close vscode, I can use finding definition again.
How can I solve it without close vscode?
![image](https://user-images.githubusercontent.com/3445470/108840655-a635d700-7611-11eb-9d16-880bf089c137.png)

**Reply by sean-mcmanus** [reply]
The cpptools or cpptools-srv processes might be stuck. We have some bug fixes with 1.2.2 that might fix this: https://github.com/microsoft/vscode-cpptools/milestone/63 . If you can attach a debugger or use profiler to get a callstack we could let you know if it's a known issue or not (https://github.com/microsoft/vscode-cpptools/wiki/Attaching-debugger-to-cpptools-or-cpptools%E2%80%90srv ).
